From 7b30871b6c82ba91e9d5550426b53a37a85b0b81 Mon Sep 17 00:00:00 2001
From: Nikhil Benesch <nikhil.benesch@gmail.com>
Date: Wed, 19 Apr 2017 14:44:25 -0400
Subject: [PATCH 15/16] arena: derive alignment unit from std::max_align_t

Using `sizeof(void*)` as the alignment unit does not guarantee suitable
alignment on all platforms. Darwin, for example, will choke when trying
to store a std::function into 8-byte-aligned memory under certain
circumstances.
---
 util/arena.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/arena.cc b/util/arena.cc
index bb5fdb8d..09eb67da 100644
--- a/util/arena.cc
+++ b/util/arena.cc
@@ -31,7 +31,7 @@ const size_t Arena::kInlineSize;
 
 const size_t Arena::kMinBlockSize = 4096;
 const size_t Arena::kMaxBlockSize = 2u << 30;
-static const int kAlignUnit = sizeof(void*);
+static const int kAlignUnit = alignof(std::max_align_t);
 
 size_t OptimizeBlockSize(size_t block_size) {
   // Make sure block_size is in optimal range
-- 
2.12.1

